cloud/misc/rhevm.py

```diff
@@ -0,0 +1,1097 @@
+            vmf['error'] = str(e)
+            vmf['failed'] = True
+    else:
+        vmf['msg'].append('VM delete protection was already set to ' + str(vmf['del_prot']))
+    return vmf
+
+
+def setBootOrder(c, vm, vmf):
+    bootorder = []
+    for device in vmf['boot_order']:
+        bootorder.append(params.Boot(dev=device))
+    if vm.os.boot != bootorder:
+        vm.os.boot = bootorder
+        try:
+            vm.update()
+            vmf['msg'].append('The boot order has been set')
+        except Exception as e:
+            vmf['msg'].append("Failed to set the boot order")
+            vmf['error'] = str(e)
+            vmf['failed'] = True
+    else:
+        vmf['msg'].append('The boot order has already been set')
+    return vmf
+
+
+def setMemory(c, vm, vmf):
+    if int(vm.memory) != int(int(vmf['vmmem']) * 1024 * 1024 * 1024):
+        # first set guaranteed memory, otherwise shrinking a vm fails
+        try:
+            memory_policy = vm.memory_policy.guaranteed
+            if int(memory_policy) != int(float(vmf['mempol']) * 1024 * 1024 * 1024):
+                vm.memory_policy.guaranteed = int(float(vmf['mempol']) * 1024 * 1024 * 1024)
+                vmf = stopVM(c, vmf)
+                if vmf['failed']:
+                    return vmf
+                vm.update()
+                vmf['changed'] = True
+                vmf['msg'].append('Memory Policy has been set to: ' + str(vmf['mempol']) + 'GB')
+        except Exception as e:
+            vmf['msg'].append("Failed to get/set the current memory policy")
+            vmf['error'] = str(e)
+            vmf['failed'] = True
+            return vmf
+
+        try:
+            vmf = stopVM(c, vmf)
+            if vmf['failed']:
+                return vmf
+            vm.memory = int(int(vmf['vmmem']) * 1024 * 1024 * 1024)
+            vm.update()
+            vmf['changed'] = True
+            vmf['changes'].append('setMemory')
+            vmf['msg'].append('Memory has been set to: ' + vmf['vmmem'] + 'GB')
+        except Exception as e:
+            vmf['msg'].append("Failed to get/set the current memory")
+            vmf['error'] = str(e)
+            vmf['failed'] = True
+            return vmf
+    return vmf
+
+
+def stopVM(c, vmf):
+    vmf = power(c, vmf, 'down')
+    return vmf
+
+
+def startVM(c, vmf):
+    vmf = power(c, vmf, 'up')
+    return vmf    
+
+
+def restartVM(c, vmf):
+    vmf = stopVM(c, vmf)
+    if vmf['failed'] is False:
+        vmf = startVM(c, vmf)
+    return vmf
+
+
+def power(connection, vmf, state):
+    try:
+        vm = connection.vms.get(name=vmf['name'])
+        if state == vm.status.state:
+            vmf['msg'].append("VM state was already " + vm.status.state)
+        else:
+            if state == "up":
+                vm.start()
+                vmf['changed'] = True
+            elif state == "down":
+                vm.stop()
+                vmf['changed'] = True
+            timeout = vmf['timeout']

```

Comment:
In order to handle timeout more easily, maybe `timeout` could be passed to the (API)[https://github.com/oVirt/ovirt-engine-sdk/blob/master/src/ovirtsdk/api.py#L78] in `connection` method ?